### PR TITLE
test(plugins) add buffering combinations iterator

### PR DESCRIPTION
**Pull request by javierguerragiraldez**
Because we all want to spend 4x times on tests...

### CODE REVIEW

**Review by kikito — APPROVED** [review]
This looks ok to me (assuming the tests pass).

Note for other reviewers: I strongly suggest reviewing this pr with the cog menu (⚙️▾) and then "✅Hide whitespace changes" option. Makes all the difference.

File: spec/helpers.lua [hunk 1]
```diff
@@ -293,6 +293,30 @@ do
   end
 end
 
+local each_buffering
```
Comment (Tieske) [inline spec/helpers.lua:296]: documentation is missing here

**Review by Tieske — CHANGES_REQUESTED** [review]
these are 2 additional layers of permutations for testing, can we not replace them with a single layer by using dynamic permutations?

for example;
```lua
local permute_bool = { true, false }

for strategy, buf_req, buf_res in permute({helpers.strategies, permute_bool, permute_bool }) do

  -- test stuff

end
```

**Comment by Tieske** [comment]
I thought Penlight had a function for this, turned out not; https://github.com/lunarmodules/Penlight/pull/360

**Comment by Tieske** [comment]
@javierguerragiraldez how to proceed? I can push a Penlight version with the permutations iterator if you want.

**Comment by javierguerragiraldez** [comment]
> @javierguerragiraldez how to proceed? I can push a Penlight version with the permutations iterator if you want.

Thanks, no need to.  this PR won't be merged.  it's more an experiment to test that the new buffering cases don't crash anything catastrophically.

i'm closing this, as it has already proved the point.

**Pull request closed without merging**